themes/classic/_dev/js/cart.js

```diff
@@ -168,6 +168,39 @@ $(document).ready(() => {
     });
   };
 
+  //Hide 'have a code promo' text if displayed when 'close' is
+  let isPresent = $('.promo-code-button').length,
+      nameLength = 0;
+
+  function labelCodeDisplayed() {
+    $('.promo-code-button').not('.cancel-promo').hide();
+    
+    $('.cancel-promo').on('click',  (e) => {
+      $(e.currentTarget).hide();
+      $('.promo-code-button:not(.cancel-promo)').show();
+    });
+
+    $('.promo-code-button:not(.cancel-promo)').on('click', (e) => {
+        $(e.currentTarget).hide();
+        $('.cancel-promo').show();
+    });
+  }
+
+  setInterval( function () {
+    const promoNameLength = $('.promo-name').length;
+
+    if (promoNameLength != nameLength) {
+      nameLength = promoNameLength;
+      labelCodeDisplayed();
+    }
+  }, 100);
+
+  $('.promo-code-button').each(() => {
```

Comment:
Why is this line needed?

Reply:
at least that one is useless. I remove it